### PR TITLE
[Ldap] Fixed CS and hardened some code

### PR DESCRIPTION
**Pull request by csarrazi**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16665 |
| License | MIT |
| Doc PR | no |

This PR takes into account remaining comments from #16665

### CODE REVIEW

**Comment by fabpot** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @csarrazi.

File: src/Symfony/Component/Ldap/Adapter/ExtLdap/Collection.php [hunk 1]
```diff
@@ -84,7 +85,13 @@ private function initialize()
             return;
         }
 
-        $entries = ldap_get_entries($this->connection->getResource(), $this->search->getResource());
+        $con = $this->connection->getResource();
```
Comment (xabbuh) [inline src/Symfony/Component/Ldap/Adapter/ExtLdap/Collection.php]: What if `getResource()` returns `null` (maybe I simply miss the point where we ensure that the connection is always established before this code is executed)?

Comment (csarrazi) [inline src/Symfony/Component/Ldap/Adapter/ExtLdap/Collection.php]: The connection is always established before this code is executed.

Either the `Query` class creates the connection if non-existent (with default values), or it uses the existing connection.

**Pull request merged**